packages/indexer/lib/index.js

```diff
@@ -198,116 +198,92 @@ class TransactionCollector {
     if (!lock && !type) {
       throw new Error("Either lock or type script must be provided!");
     }
-    if (lock) {
+    // Wrap the plain `Script` to `ScriptWrapper`.
+    if (lock && !lock.script) {
       validators.ValidateScript(lock);
+      this.lock = { script: lock, io_type: "both" };
+    }
+    if (lock && lock.script) {
+      validators.ValidateScript(lock.script);
+      this.lock = lock;
     }
-    if (type) {
+    if (type && !type.script) {
       validators.ValidateScript(type);
+      this.lock = { script: type, io_type: "both" };
+    }
+
+    if (type && type.script) {
+      validators.ValidateScript(type.script);
+      this.type = type;
     }
+
     this.indexer = indexer;
-    this.lock = lock;
-    this.type = type;
     this.skipMissing = skipMissing;
     this.includeStatus = includeStatus;
     this.fromBlock = fromBlock;
     this.toBlock = toBlock;
     this.rpc = new RPC(indexer.uri);
   }
 
-  async count() {
-    if (this.lock && this.type) {
-      const lockHashes = new OrderedSet(
+  get_transaction_hashes() {
+    let hashes = null;
+    let lockHashes = null;
+    let typeHashes = null;
+
+    if (this.lock) {
+      let script_type = 0;
+      lockHashes = new OrderedSet(
         this.indexer
           ._getTransactionsByScriptIterator(
-            this.lock,
-            0,
+            this.lock.script,
+            script_type,
```

Comment:
Why give this a variable if you will never modify/use it?

Reply:
I think give it an identifier is more readable than raw `0|1` , and yes, it should be a const type here. Will fix it.